skll/experiments.py

```diff
@@ -791,89 +792,95 @@ def run_configuration(config_file, local=False, overwrite=True, queue='all.q',
-            job_args["learner_name"] = learner_name
-            job_args["train_path"] = train_path
-            job_args["test_path"] = test_path
-            job_args["train_set_name"] = train_set_name
-            job_args["test_set_name"] = test_set_name
-            job_args["shuffle"] = do_shuffle
-            job_args["model_path"] = model_path
-            job_args["prediction_prefix"] = prediction_prefix
-            job_args["grid_search"] = do_grid_search
-            job_args["grid_objective"] = grid_objective
-            job_args["suffix"] = suffix
-            job_args["log_path"] = temp_logfile
-            job_args["probability"] = probability
-            job_args["results_path"] = results_path
-            job_args["sampler_parameters"] = (fixed_sampler_parameters
-                                              if fixed_sampler_parameters
-                                              else dict())
-            job_args["fixed_parameters"] = (fixed_parameter_list[learner_num]
-                                            if fixed_parameter_list
-                                            else dict())
-            job_args["param_grid"] = (param_grid_list[learner_num]
-                                      if param_grid_list else None)
-            job_args["pos_label_str"] = pos_label_str
-            job_args["overwrite"] = overwrite
-            job_args["feature_scaling"] = feature_scaling
-            job_args["min_feature_count"] = min_feature_count
-            job_args["grid_search_jobs"] = grid_search_jobs
-            job_args["grid_search_folds"] = grid_search_folds
-            job_args["cv_folds"] = cv_folds
-            job_args["save_cv_folds"] = save_cv_folds
-            job_args["do_stratified_folds"] = do_stratified_folds
-            job_args["label_col"] = label_col
-            job_args["id_col"] = id_col
-            job_args["ids_to_floats"] = ids_to_floats
-            job_args["quiet"] = quiet
-            job_args["class_map"] = class_map
-            job_args["custom_learner_path"] = custom_learner_path
-
-            if not local:
-                jobs.append(Job(_classify_featureset, [job_args],
-                                num_slots=(MAX_CONCURRENT_PROCESSES if
-                                           do_grid_search else 1),
-                                name=job_name, queue=queue))
-            else:
-                _classify_featureset(job_args)
+            for grid_objective in grid_objectives:
+
+                # for the individual job name, we need to add the feature set name
+                # and the learner name
+                if len(grid_objectives) == 1:
+                    job_name_components = [experiment_name, featureset_name,
+                                           learner_name]
+                else: 
+                    job_name_components = [experiment_name, featureset_name,
+                                       learner_name, grid_objective]
```

Comment:
Nitpick:  Please align `learner_name` with the opening bracket on this line (like is done three lines above).